db/blob/blob_file_builder.h

```diff
@@ -79,6 +81,7 @@ class BlobFileBuilder {
   std::string column_family_name_;
   Env::IOPriority io_priority_;
   Env::WriteLifeTimeHint write_hint_;
+  std::vector<std::string>* blob_file_paths_;
```

Comment:
Nit: since we assert `blob_file_paths_.empty() && blob_file_additions_.empty()`, would it be easier to make these two members belong to `BlobFileBuilder`?
Furthermore, in non-debug mode, the assertion becomes no-op.

Reply:
The reason I didn't make these members is that they outlive `BlobFileBuilder` and I wanted to avoid copying them (or having to introduce unconventional "read once" type of accessors that transfer them to the caller).